### PR TITLE
#51 storybook

### LINKED ISSUE #51: Add storybook
Before creating so many packages, would it be useful to have story book set up?

I know when working in a big code base like Grafana it was incredibly useful.

I know this will take some time to setup. Will take more time if we want to set it up.. later?

@mark-prins think this is useful?

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #51 

image

cooooool

- I left the default `stories` folder for now because it's easier than looking at the docs..
- if we have a `test` and a `story` and an actual component file for each component we might need a better structure... unsure if we should just use the `stories` folder and put all the stories in there or if having a folder per component with the test and story in is nice also.. :(
- I added `react` and `react-dom` to dev deps of the root as story book needed it there.. when I added them react 17 was added.. so I just updated all of them.. mighhhht need to `rm -rf node_modules` and `yarn install`..

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  stories: [
+    '../packages/**/*.stories.mdx',
+    '../packages/**/*.stories.@(js|jsx|ts|tsx)',
+  ],
```
Comment (joshxg) [inline .storybook/main.js:5]: unsure where to actually put stories

File: .storybook/theme.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  barTextColor: '#f2f2f5',
+  barSelectedColor: 'black',
+  barBg: '#e95c30',
+
+  // Form colors
+  inputBg: 'blue',
+  inputBorder: 'blue',
+  inputTextColor: 'blue',
+  inputBorderRadius: 4,
+});
```
Comment (joshxg) [inline .storybook/theme.js:36]: I was just messing around it was a bit of fun. Feel free to tweak any values you want

File: package.json [hunk 3]
```diff
@@ -8,14 +8,23 @@
     "compile": "lerna run --scope @openmsupply-client/* --parallel tsc",
     "prepare": "husky install",
     "pre-commit-lint": "yarn compile && yarn lint-staged",
-    "test": "jest --config ./jest.config.js"
+    "test": "jest --config ./jest.config.js",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook"
```
Comment (joshxg) [inline package.json:13]: unsure if we should use a `@openmsupply-client/ui` package with all the building blocks and just use story book for that..

But on the other hand, it is a super nice way for 'visual testing' - easily be able to switch through components and check that strings fit etc.

File: package.json [hunk 4]
```diff
@@ -8,14 +8,23 @@
...
     ]
   },
   "devDependencies": {
+    "@babel/core": "^7.15.0",
+    "@storybook/addon-actions": "^6.3.7",
+    "@storybook/addon-essentials": "^6.3.7",
+    "@storybook/addon-links": "^6.3.7",
+    "@storybook/addons": "^6.3.7",
+    "@storybook/react": "^6.3.7",
+    "@storybook/theming": "^6.3.7",
```
Comment (joshxg) [inline package.json:27]: thats a lot!

File: packages/common/src/ui/components/LoadingApp.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import { LoadingApp } from '.';
+
+const Template: ComponentStory<typeof LoadingApp> = () => <LoadingApp />;
+
+export const Primary = Template.bind({});
+
+export default {
+  title: 'Example/Loading',
+  component: LoadingApp,
+} as ComponentMeta<typeof LoadingApp>;
```
Comment (joshxg) [inline packages/common/src/ui/components/LoadingApp.stories.tsx:12]: simple example

File: packages/common/src/utils/testing.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  locale = 'en',
+}) => {
+  return (
+    <AppThemeProvider>
+      <IntlTestProvider locale={locale}>
+        <BrowserRouter>{children}</BrowserRouter>
+      </IntlTestProvider>
+    </AppThemeProvider>
+  );
+};
```
Comment (joshxg) [inline packages/common/src/utils/testing.tsx:22]: a lot of stuff will need this so makes writing tests/stories easier.. not sure if this is a good place for it

File: packages/host/src/AppDrawer.tsx [hunk 7]
```diff
@@ -22,7 +22,7 @@ import {
 import clsx from 'clsx';
 
 const CustomersNav = React.lazy(() =>
-  process.env['NODE_ENV'] === 'test'
+  process.env['NODE_ENV'] !== 'production'
```
Comment (joshxg) [inline packages/host/src/AppDrawer.tsx:25]: 🏃

File: stories/Button.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
```
Comment (joshxg) [inline stories/Button.stories.tsx:1]: I left all these in to reference.. can remove?

File: .storybook/theme.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  colorSecondary: '#e95c30',
+
+  // UI
+  appBg: '#555770',
+  appContentBg: 'white',
+  appBorderColor: 'white',
+  appBorderRadius: 4,
+
+  // Typography
+  fontBase: '"Open Sans", sans-serif',
```
Comment (mark-prins) [inline .storybook/theme.js:16]: I spent a while trying to 'fix' this and bring in a better font.. but have given up.

File: .storybook/theme.js (hunk 2 above)
Comment (mark-prins) [reply .storybook/theme.js:36]: maybe later 😉  all good.

File: package.json (hunk 3 above)
Comment (mark-prins) [reply package.json:13]: I really like the idea of a ui package but do worry that updating, bumping revisions, updating usages.. all too much hassle at the mo. maybe later?
agree though, that storybook does help with checking out components which are built of items in the ui package, and so it is good to have it at this level

File: packages/common/src/utils/testing.tsx (hunk 6 above)
Comment (mark-prins) [reply packages/common/src/utils/testing.tsx:22]: nice! not sure either.. seems ok to me

File: stories/Button.stories.tsx (hunk 8 above)
Comment (mark-prins) [reply stories/Button.stories.tsx:1]: can we comment somewhere to say that they are here for reference? rename the folder?
agree that they're good to have, just threw me at first, wondering why there were stories here, and next to components

**Review by mark-prins — APPROVED** [review]
badass!
am now keen to get it in and create stories for shared components and stuff like the appbar to show the options available.

File: .storybook/main.js (hunk 1 above)
Comment (joshxg) [reply .storybook/main.js:5]: Think you're suggesting that having the stories at the same level is no good?

**Comment by joshxg** [comment]
Merging into my other pr to do the tidy ups there.

**Pull request merged**